GitHubRepositories.ps1

```diff
@@ -1607,3 +1607,530 @@ function Move-GitHubRepositoryOwnership
+
+    .INPUTS
+        None
+
+    .OUTPUTS
+        PSCustomObject
+
+    .EXAMPLE
+        Get-GitHubRepositoryVulnerabilityAlert -OwnerName Microsoft -RepositoryName PowerShellForGitHub
+
+        Retrieves the status of vulnerability alerts for the PowerShellForGithub repository.
+    .EXAMPLE
+        Get-GitHubRepositoryVulnerabilityAlert -Uri https://github.com/PowerShell/PowerShellForGitHub
+
+        Retrieves the status of vulnerability alerts for the PowerShellForGithub repository.
+#>
+    [CmdletBinding(
+        PositionalBinding = $false,
+        DefaultParameterSetName='Elements')]
+    param(
+        [Parameter(ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            Position = 1,
+            ParameterSetName='Uri')]
+        [string] $Uri,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog -Invocation $MyInvocation
+
+    $elements = Resolve-RepositoryElements -BoundParameters $PSBoundParameters
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $params = @{
+        UriFragment = "repos/$OwnerName/$RepositoryName/vulnerability-alerts"
+        Description =  "Getting Vulnerability Alerts status for $RepositoryName"
+        AcceptHeader = $script:dorianAcceptHeader
+        Method = 'Get'
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+        NoStatus = (Resolve-ParameterWithDefaultConfigurationValue -BoundParameters $PSBoundParameters `
+            -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    try
+    {
+        Invoke-GHRestMethod @params | Out-Null
+        $result = $true
+    }
+    catch
+    {
+        # Temporary code to handle current differences in exception object between PS5 and PS7
+        if ($PSVersionTable.PSedition -eq 'Core')
+        {
+            if ($_.Exception -is [Microsoft.PowerShell.Commands.HttpResponseException] -and
+                ($_.ErrorDetails.Message | ConvertFrom-Json).message -eq 'Vulnerability alerts are disabled.')
+            {
+                $result = $false
+            }
+            else
+            {
+                throw $_
+            }
+        }
+        else
+        {
+            if ($_.Exception.Message -like '*Vulnerability alerts are disabled.*')
+            {
+                $result = $false
+            }
+            else
+            {
+                throw $_
+            }
+        }
+    }
+
+    return $result
+}
+
+Function Enable-GitHubRepositoryVulnerabilityAlert {
```

Comment:
Please update all of these to be filters with pipeline support.  Can you also move the `{` to the next line? Here and elsewhere.

Reply:
Done